src/components/manage/Controlpanels/AddonsControlpanel.jsx

```diff
@@ -0,0 +1,384 @@
+    availableAddons: state.addons.availableAddons,
+    upgradableAddons: state.addons.upgradableAddons,
+    activeIndex: state.activeIndex,
+    pathname: props.location.pathname,
+  }),
+  dispatch => bindActionCreators({ installAddon, listAddons, uninstallAddon }, dispatch),
+)
+/**
+ * AddonsControlpanel class.
+ * @class AddonsControlpanel
+ * @extends Component
+ */
+export default class AddonsControlpanel extends Component {
+  /**
+   * Property types.
+   * @property {Object} propTypes Property types.
+   * @static
+   */
+  static propTypes = {
+    listAddons: PropTypes.func.isRequired,
+    installAddon: PropTypes.func.isRequired,
+    uninstallAddon: PropTypes.func.isRequired,
+    installedAddons: PropTypes.arrayOf(
+      PropTypes.shape({
+        '@id': PropTypes.string,
+        'id': PropTypes.string,
+        'title': PropTypes.string,
+        'version': PropTypes.string,
+        'description': PropTypes.string,
+        'upgrade_info': PropTypes.shape({
+          'available': PropTypes.boolean
+        })
+      })
+    ).isRequired,
+    availableAddons: PropTypes.arrayOf(
+      PropTypes.shape({
+        '@id': PropTypes.string,
+        'id': PropTypes.string,
+        'title': PropTypes.string,
+        'version': PropTypes.string,
+        'description': PropTypes.string,
+        'upgrade_info': PropTypes.shape({
+          'available': PropTypes.boolean
+        })
+      })
+    ).isRequired,
+    upgradableAddons: PropTypes.arrayOf(
+      PropTypes.shape({
+        '@id': PropTypes.string,
+        'id': PropTypes.string,
+        'title': PropTypes.string,
+        'version': PropTypes.string,
+        'description': PropTypes.string,
+        'upgrade_info': PropTypes.shape({
+          'available': PropTypes.boolean
+        })
+      })
+    ).isRequired,
+  };
+
+  /**
+   * Constructor
+   * @method constructor
+   * @param {Object} props Component properties
+   * @constructs Sharing
+   */
+  constructor(props) {
+    super(props);
+    this.onAccordionClick = this.onAccordionClick.bind(this);
+    this.onInstall = this.onInstall.bind(this);
+    this.onUninstall = this.onUninstall.bind(this);
+    this.state = {
+      activeIndex: -1,
+      installedAddons: [],
+      availableAddons: [],
+      upgradableAddons: [],
+    };
+  }
+
+  /**
+   * On accordion click handler
+   * @method onAccordionClick
+   * @param {object} event Event object.
+   * @returns {undefined}
+   */
+  onAccordionClick(event, titleProps) {
+    const { index } = titleProps
+    const { activeIndex } = this.state.activeIndex
+    const newIndex = activeIndex === index ? -1 : index
+
+    this.setState({ activeIndex: newIndex })
+  }
+
+  /**
+   * Install handler
+   * @method onInstall
+   * @param {Object} event Event object.
+   * @param {string} value Install value.
+   * @returns {undefined}
+   */
+  onInstall(event, { value }) {
+    event.preventDefault();
+    this.props.installAddon(value);
+    this.render();
```

Comment:
This should not be needed, if you call the `installAddon` action that should result in some change in the store so it will trigger the render automatically since the props change. What you can do is map the `loading` and `loaded` state of the `installAddon` action in this component and then check for those in the `componentWillReceiveProps` to refetch the list of installed add-ons. You can check `components/manage/Display/Display.jsx` for an example.